migration/migrator/migrations/course/20220612122726_gradeable_specific_anon_ids.py

```diff
@@ -36,28 +36,22 @@ def up(config, database, semester, course):
         db_info = json.load(db_file, object_pairs_hook=OrderedDict)
         database.execute("GRANT SELECT, INSERT, UPDATE, DELETE ON gradeable_anon TO {}".format(db_info['database_course_user']))
 
-    database.execute(
-        """
-        INSERT INTO gradeable_anon (
-            SELECT u.user_id, g_id, u.anon_id
-            FROM gradeable g JOIN users u ON 1=1 WHERE NOT EXISTS (SELECT 1 FROM gradeable_anon WHERE user_id=u.user_id AND g_id=g.g_id)
-        );
-        """
-    )
-    database.execute("ALTER TABLE users DROP COLUMN anon_id;")
+    database.execute("""
+        DO
+        $do$
+        BEGIN
+        IF EXISTS (SELECT column_name
+                   FROM information_schema.columns
+                   WHERE table_name='users' and column_name='anon_id') THEN
+               INSERT INTO gradeable_anon (
+                    SELECT u.user_id, g_id, u.anon_id
+                    FROM gradeable g JOIN users u ON 1=1 WHERE NOT EXISTS (SELECT 1 FROM gradeable_anon WHERE user_id=u.user_id AND g_id=g.g_id) AND u.anon_id IS NOT NULL
+                );
+            END IF;
+        END
+        $do$
+    """)
+    database.execute("ALTER TABLE users DROP COLUMN IF EXISTS anon_id;")
 
 def down(config, database, semester, course):
-    """
-    Run down migration (rollback).
-
-    :param config: Object holding configuration details about Submitty
-    :type config: migrator.config.Config
-    :param database: Object for interacting with given database for environment
-    :type database: migrator.db.Database
-    :param semester: Semester of the course being migrated
-    :type semester: str
-    :param course: Code of course being migrated
-    :type course: str
-    """
-    database.execute("ALTER TABLE users ADD COLUMN IF NOT EXISTS anon_id character varying(255);")
-    database.execute("UPDATE users u SET anon_id=(SELECT ga.anon_id FROM gradeable_anon ga WHERE ga.user_id=u.user_id LIMIT 1);")
+    pass
```

Comment:
Why remove these? If the PR gets reverted for some reason and this down migration was run, the old code wouldn't work.

Reply:
I was not feeling sure about side-effects of running down migration, as gradeable specific `anon_id` could be used in file paths and switching back to old model would use different `anon_id` than what was used to create the file. But not being able to use Submitty at all is a bigger problem than some less probable side-effects, so, I've added it back.